src/realm.js

```diff
@@ -117,10 +117,12 @@ function initRootRealm(parentUnsafeRec, self, options) {
  * A compartment shares the intrinsics of its root realm. Here, only a
  * realmRec is necessary to hold the global object, eval() and Function().
  */
-function initCompartment(unsafeRec, self) {
+function initCompartment(unsafeRec, self, options) {
   // note: 'self' is the instance of the Realm.
 
-  const realmRec = createRealmRec(unsafeRec);
+  options = Object(options);
```

Comment:
Is the purpose of this to convert `undefined` to `{}` ? This will also convert, for example, a number to a Number wrapper, etc. These other conversions are likely harmless, but are probably more than we want to worry about.

If that is the purpose, I suggest making the `options` parameter explicitly optional with default
```js
function initCompartment(unsafeRec, self, options = {}) {
```
which is more self-documenting anyway.